dashboard/app/models/concerns/pd/jot_form_backed_form.rb

```diff
@@ -169,10 +177,29 @@ def _sync_from_jotform(form_ids)
         imported
       end
 
-      def process_submission(submission)
+      # Process a submission and saves the model.
+      # @param [submission] (optional) The submission received from JotForm.
+      # @param [questions_details] (optional) Details of the questions.  If provided, we can
+      #   attempt to use the name field from each question rather than its numerical ID when
+      #   storing answers.
+      def process_submission(submission, questions_details=nil)
         # There should be no duplicates, but just in case handle them gracefully as an upsert.
         find_or_initialize_by(submission.slice(:form_id, :submission_id)).tap do |model|
-          model.answers = submission[:answers].to_json
+          dest_answers = {}
+
+          source_answers = submission[:answers]
+          source_answers.each do |key, value|
+            # We have an answer ID, but can we get a name instead of the ID?
+            question_detail = questions_details && questions_details.find {|q| q["id"] == key}
```

Comment:
```suggestion
            question_detail = questions_details&.find {|q| q["id"] == key}
```

Reply:
Will address in https://github.com/code-dot-org/code-dot-org/pull/28850.